video/out/vo_gpu_next.c

```diff
@@ -1948,11 +1948,15 @@ static void update_render_options(struct vo *vo)
     pars->params.plane_upscaler = map_scaler(p, SCALER_CSCALE);
     pars->params.frame_mixer = opts->interpolation ? map_scaler(p, SCALER_TSCALE) : NULL;
 
-    // Request as many frames as required from the decoder
+    // Request as many frames as required from the decoder, depending on the
+    // speed VPS/FPS ratio libplacebo may need more frames. Request frames up to
+    // ratio of 1/4, but only if anti aliasing is enabled.
     int req_frames = 2;
-    if (pars->params.frame_mixer)
-        req_frames += ceilf(pars->params.frame_mixer->kernel->radius);
-    vo_set_queue_params(vo, 0, req_frames);
+    if (pars->params.frame_mixer) {
+        req_frames += ceilf(pars->params.frame_mixer->kernel->radius) *
+                      (pars->params.skip_anti_aliasing ? 1 : 4);
```

Comment:
Isn't this always requesting 4x more frames than needed, even when VPS >= FPS? Can't we do this better?

One possible design is to just bump up the limit if we ever see PL_QUEUE_MORE.

Reply:
Ideally we would adjust the limit before requesting the frames, else we will produce underrun on first try and possibly every next. And it is not clear when to reduce the limit.

Do you see an issue with bigger limit? I didn't have good idea to make it smarter than just requesting more frames up front.